python/sparkdl/transformers/tf_tensor.py

```diff
@@ -69,37 +42,53 @@ class TFOneDimTensorTransformer(Transformer, HasInputCol, HasOutputCol,
       dimension) in the tensor shapes.
-    - Input DataFrame has an array column where all elements have the same length    
+    - Input DataFrame has an array column where all elements have the same length
     """
-    
+
     @keyword_only
-    def __init__(self, tfGraph=None, inputTensor=None, outputTensor=None,
-                 inputCol=None, outputCol=None):
+    def __init__(self, inputMapping=None, outputMapping=None, tfGraph=None, hparams=None):
         """
-        __init__(self, tfGraph=None, inputTensor=None, outputTensor=None,
-                 inputCol=None, outputCol=None)
+        __init__(self, inputMapping=None, outputMapping=None, tfGraph=None, hparams=None)
         """
-        super(TFOneDimTensorTransformer, self).__init__()
+        super(TFTensorTransformer, self).__init__()
         kwargs = self._input_kwargs
         self.setParams(**kwargs)
 
     @keyword_only
-    def setParams(self, tfGraph=None, inputTensor=None, outputTensor=None,
-                  inputCol=None, outputCol=None):
+    def setParams(self, inputMapping=None, outputMapping=None, tfGraph=None, hparams=None):
         """
-        setParams(self, tfGraph=None, inputTensor=None, outputTensor=None,
-                  inputCol=None, outputCol=None)
+        setParams(self, inputMapping=None, outputMapping=None, tfGraph=None, hparams=None)
         """
-        super(TFOneDimTensorTransformer, self).__init__()
+        super(TFTensorTransformer, self).__init__()
         kwargs = self._input_kwargs
         return self._set(**kwargs)
-    
-    def _transform(self, df):        
+
+    def _transform(self, dataset):
+        df = dataset
+        output_renaming = {}
+
         with IsolatedSession(graph=self.getTFGraph()) as issn:
-            tnsr_in = tfx.get_tensor(issn.graph, self.getInputTensor())
-            tnsr_out = tfx.get_tensor(issn.graph, self.getOutputTensor())
-            assert len(tnsr_out.shape) >= 1, \
-                'output tensor must have a leading dimension: {}'.format(tnsr_out.shape)
-            gfn = issn.asGraphFunction([tnsr_in], [tnsr_out])            
-        
-        return _tranform_with_graph_function(gfn, df, self.getInputCol(), self.getOutputCol())
+            feeds = []
+            for colname, tnsr in self.getInputMapping().items():
+                feeds.append(tfx.get_tensor(issn.graph, tnsr))
+                new_colname = tfx.op_name(issn.graph, tnsr)
+                df = df.withColumnRenamed(colname, new_colname)
+
+            fetches = []
+            for tnsr, new_colname in self.getOutputMapping().items():
+                fetches.append(tfx.get_tensor(issn.graph, tnsr))
+                old_colname = tfx.op_name(issn.graph, tnsr)
+                output_renaming[old_colname] = new_colname
+
+            gfn = issn.asGraphFunction(feeds, fetches)
+
+        analyzed_df = tfs.analyze(df)
+
+        with IsolatedSession() as issn:
+            _, fetches = issn.importGraphFunction(gfn, prefix='')
```

Comment:
It also seems that importing graph_def directly is a bit flaky. `importGraphFunction` attaches the names and tensors correctly. It might just save us a few lines of code.